types/ember__component/helper.d.ts

```diff
@@ -71,14 +84,16 @@ interface LegacyArgsFor<T> {
 // all `ExpandSignature` types fully general to work with *any* invokable. But
 // "future" here probably means Ember v5. :sobbing:
 export interface ExpandSignature<T> {
-    Args: keyof T extends 'Args' | 'Return' // Is this a `Signature`?
+    Args: unknown extends T // Is this the default (i.e. unspecified) signature?
+        ? MissingSignatureArgs // Then return our special "missing signature" type
+        : keyof T extends 'Args' | 'Return' // Is this a `Signature`?
         ? ArgsFor<T> // Then use `Signature` args
         : LegacyArgsFor<T>; // Otherwise fall back to classic `Args`.
     Return: 'Return' extends keyof T ? T['Return'] : unknown;
 }
 
-type NamedArgs<S> = ExpandSignature<S>['Args']['Named'];
-type PositionalArgs<S> = ExpandSignature<S>['Args']['Positional'];
+type NamedArgs<S> = unknown extends S ? Record<string, unknown> : ExpandSignature<S>['Args']['Named'];
```

Comment:
For the sake of future us, maybe a comment here explaining that the `unknown extends S` checks are

```suggestion
// The `unknown extends S` checks on both of these are here to preserve backward
// compatibility with the existing non-`Signature` definition. When migrating
// into Ember or otherwise making a breaking change, we can drop the "default"
// in favor of just using `ExpandSignature`.
type NamedArgs<S> = unknown extends S ? Record<string, unknown> : ExpandSignature<S>['Args']['Named'];
```